include/common/enum.inc

```diff
@@ -842,7 +840,37 @@ R(FuncRef, 0x70)
 R(ExternRef, 0x6F)
 
 #undef R
-#endif // UseRefType
+#endif // UseRefType 
+
+#ifdef UseInterfaceType
+#define In Line 
+
+//In(Unit, 0x7f, "unit")
+In(Bool, 0x7e, "bool")
+In(S8, 0x7d, "s8")
+In(U8, 0x7c, "u8")
+In(S16, 0x7b, "s16")
+In(U16, 0x7a, "u16")
+In(S32, 0x79, "s32")
+In(U32, 0x78, "u32")
+In(S64, 0x77, "s64")
+In(U64, 0x76, "u64")
+In(Float32, 0x75, "float32")
+In(Float64, 0x74, "float64")
+In(Char, 0x73, "char")
+In(String, 0x72, "string")
+In(Record, 0x71, "record")
+In(Variants, 0x70, "variants")
+//In(List, 0x6f, "list")
+In(Tuple, 0x6e, "tuple")
+In(Flag, 0x6d, "flag") 
+In(Enum, 0x6c, "enum")
+In(Union, 0x6b, "union")
+//In(Option, 0x6a, "option")
```

Comment:
I'm currently thinking of the Implementation for these types similar like `Record` and others . As I'm not sure about what it would be for the `Unit` type? as it accepts everything.

Reply:
Okay ig `Unit` will equivalent to `void` .